src/iframe-attributes.js

```diff
@@ -57,6 +60,14 @@ export function getContextMetadata(
       href: locationHref,
     },
     startTime,
+    tagName: element.tagName,
+    mode: getModeObject(),
+    canary: isCanary(parentWindow),
+    hidden: !viewer.isVisible(),
+    initialIntersection: element.getIntersectionChangeEntry ?
```

Comment:
Fixed the issue. I had thought that the A4A path didn't have the getIntersectionChangeEntry because in all of our tests the element didn't have it. But it turns out that this way because our tests weren't really making an AMP element, they were just making a dummy HTML element with *some* amount of AMP element methods attached to it. Didn't take that long to realize the test itself was what was broken, but took me and Taymon a fair while to actually get to the root cause. 

Thanks for catching my error here, Hongfei! Showed the team some potential issues with our testing that we are going to try to fix as soon as we can.